helm-charts/seldon-benchmark-workflow/templates/workflow.yaml

```diff
@@ -0,0 +1,450 @@
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  {{- if eq .Values.workflow.useNameAsGenerateName "false" }}
+  name: {{ .Values.workflow.name }}
+  {{- else }}
+  generateName: {{ .Values.workflow.name }}
+  {{- end }}
+  namespace: {{ .Values.workflow.namespace }}
+spec:
+  entrypoint: seldon-benchmark-process
+  templates:
+  - name: seldon-benchmark-process
+    parallelism: {{ .Values.workflow.parallelism }}
+    steps:
+    - - name: generate-parameters
+        template: generate-parameters-template
+    - - name: run-benchmark-iteration
+        arguments:
+          parameters:
+          - name: listitem
+            value: "{{`{{item.name}}`}}"
+          - name: replicas
+            value: "{{`{{item.replicas}}`}}"
+          - name: serverWorkers
+            value: "{{`{{item.serverWorkers}}`}}"
+          - name: serverThreads
+            value: "{{`{{item.serverThreads}}`}}"
+          - name: modelUri
+            value: "{{`{{item.modelUri}}`}}"
+          - name: image
+            value: "{{`{{item.image}}`}}"
+          - name: server
+            value: "{{`{{item.server}}`}}"
+          - name: apiType
+            value: "{{`{{item.apiType}}`}}"
+          - name: requestsCpu
+            value: "{{`{{item.requestsCpu}}`}}"
+          - name: requestsMemory
+            value: "{{`{{item.requestsMemory}}`}}"
+          - name: limitsCpu
+            value: "{{`{{item.limitsCpu}}`}}"
+          - name: limitsMemory
+            value: "{{`{{item.limitsMemory}}`}}"
+          - name: benchmarkCpu
+            value: "{{`{{item.benchmarkCpu}}`}}"
+          - name: concurrency
+            value: "{{`{{item.concurrency}}`}}"
+          - name: duration
+            value: "{{`{{item.duration}}`}}"
+          - name: rate
+            value: "{{`{{item.rate}}`}}"
+          - name: params
+            value: "{{`{{item.params}}`}}"
+          - name: disableOrchestrator
+            value: "{{`{{item.disableOrchestrator}}`}}"
+        withParam: "{{`{{steps.generate-parameters.outputs.result}}`}}"
+        template: run-benchmark-iteration-step-template
+
+  - name: run-benchmark-iteration-step-template
+    inputs:
+      parameters:
```

Comment:
I don't think listing the parameters here again is required - I have the feeling these can be just listed once globally for the workflow.

Reply:
unfortunately it is needed, each of the sections is like a "function", so you have to define the arguemnts, and then pass them, at every level